### PR TITLE
refactor(stat): Move to clap

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

**Comment by sylvestre** [comment]
I am planning to land this Thursday 5st if no objection!

File: src/uu/stat/src/stat.rs [hunk 1]
```diff
@@ -86,9 +86,17 @@ macro_rules! print_adjusted {
     };
 }
 
-static NAME: &str = "stat";
+static ABOUT: &str = "display file or file system status.";
```
Comment (rivy) [inline src/uu/stat/src/stat.rs:89]:
```suggestion
static ABOUT: &str = "Display file or file system status.";
```

**Review by rivy — CHANGES_REQUESTED** [review]
One minor capitalization nit and needs `cargo fmt`. Otherwise, good to go.

**Comment by sylvestre** [comment]
@rivy  Fixed! 
We don't show rusfmt warnings in the "Files changed" anymore?

**Comment by rivy** [comment]
> We don't show rusfmt warnings in the "Files changed" anymore?

😕 That's where they showed up for me... it's a GitHub UI thing, not specifically a decision on our part. From my perspective, if/when we generate a GHA compatible warning with file and line number, it is showing up next to those lines when I look into the "Files changed" section. IMO, It's a helpful convenience since we can't really visibly surface warnings easily elsewhere at the moment.

**Comment by sylvestre** [comment]
Bizarre, maybe I just missed it :)

**Review by rivy — APPROVED** [review]

**Pull request merged**